analytics/tests/test_views.py

```diff
@@ -714,6 +714,29 @@ def test_activate_or_deactivate_realm(self) -> None:
             m.assert_called_once_with(lear_realm)
             self.assert_in_success_response(["Realm reactivation email sent to admins of lear"], result)
 
+    def test_change_subdomain(self) -> None:
+        cordelia = self.example_user('cordelia')
+        lear_realm = get_realm('lear')
+        self.login_user(cordelia)
+        current_subdomain = lear_realm.string_id
+
+        result = self.client_post("/activity/support", {"realm_id": f"{lear_realm.id}",
+                                                        "new_subdomain": "new_name"})
+        self.assertEqual(result.status_code, 302)
+        self.assertEqual(result["Location"], "/login/")
+        self.login('iago')
+
+        result = self.client_post("/activity/support", {"realm_id": f"{lear_realm.id}", "new_subdomain": "new_name"})
+        self.assert_in_success_response([f"Subdomain changed from {current_subdomain} to new_name"], result)
+        lear_realm = get_realm('new_name')
+        self.assertEqual(lear_realm.string_id, "new_name")
```

Comment:
Checking the `string_id` is useless. It's obvious that the realm returned by `get_realm` will have the same string_id passed to the function call. You can check whether the `id` remains the same instead.

Reply:
Also add a check to make sure that there are no realms with the string_id "lear"